Solutions implemented

NikolayAbysov opened the pull request.

NikolayAbysov committed:
Solutions implemented

Sofasmile commented on src/main/java/mate/academy/internetshop/model/User.java:
```diff
@@ -1,20 +1,37 @@
     private String name;
     private String login;
     private String password;
+    private List<Role> roles;
```
давай краще використаємо сет тут

NikolayAbysov replied:
Hi! Changed to set

Sofasmile commented on src/main/java/mate/academy/internetshop/web/filter/AuthorizationFilter.java:
```diff
@@ -0,0 +1,77 @@
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/users/all", List.of(Role.RoleName.ADMIN));
```
думаю варто переглянути всі наявні юрл, і додати до них ролі

NikolayAbysov replied:
All URLs checked

Sofasmile commented on src/main/java/mate/academy/internetshop/web/filter/AuthorizationFilter.java:
```diff
@@ -0,0 +1,77 @@
+        User user = userService.get(userId);
+        if (isAuthorized(user, protectedUrls.get(reqUrl))) {
+            filterChain.doFilter(req, resp);
+            return;
```
чи можемо обійтись без return? І в else блоці також?

NikolayAbysov replied:
Yep:)

NikolayAbysov committed:
Issues fixed:
1 All URL's added to filter
2 List changed to Set in User model
3 Other minor changes
